hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/fs/OzoneManagerFS.java

```diff
@@ -49,6 +69,21 @@ OpenKeySession createFile(OmKeyArgs args, boolean isOverWrite,
    */
   OmKeyInfo lookupFile(OmKeyArgs args, String clientAddress) throws IOException;
 
+  /**
+   * List the status for a file or a directory and its contents.
+   *
+   * @param keyArgs       the args of the key provided by client.
+   * @param recursive     For a directory if true all the descendants of a
+   *                      particular directory are listed
+   * @param startKey      Key from which listing needs to start. If startKey
+   *                      exists its status is included in the final list.
+   * @param numEntries    Number of entries to list from the start key
+   * @param clientAddress a hint to key manager, order the datanode in returned
+   *                      pipeline by distance between client and datanode.
+   * @return list of file status
+   * @throws IOException if file or bucket or volume does not exist
+   */
   List<OzoneFileStatus> listStatus(OmKeyArgs keyArgs, boolean recursive,
-      String startKey, long numEntries) throws IOException;
+      String startKey, long numEntries, String clientAddress)
```

Comment:
In my [comment](https://github.com/apache/hadoop-ozone/pull/1192#discussion_r481025587) on the previous PR:

> Should we also keep the two original methods?

I meant both `getFileStatus` and `listStatus`.  Sorry if it was unclear.

Reply:
Sure, I will make the changes in my next commit.

Reply:
Sorry for re-opening the discussion on this comment. On a second thought `listStatus` is only used in [one place in the source code](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L2910) and there won't be any other caller for `listStatus() without ClientAddress`. So, how about deleting the original method and can add it later if there is any need ?

Reply:
> listStatus is only used in one place in the source code

It is also used by several tests, which all pass `null` as address.

Reply:
OK, addressed in my new commit